pkg/kubelet/custommetrics/support.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package custommetrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	// Key for the annotation defining custom metrics for pod.
+	// More info: http://releases.k8s.io/HEAD/docs/proposals/custom-metrics.md#api
+	CustomMetricsAnnotationKey = "metrics.alpha.kubernetes.io/custom-endpoints"
+
+	CustomMetricsDefinitionContainerPath = "/etc/cadvisor-custom-metrics-prometheus.json"
+)
+
+// CustomMetricsEndpoint is a structure defining endpoint for pod's custom metric.
+// It is parsed from JSON from the value of the annotation with CustomMetricsAnnotationKey.
+// More info: http://releases.k8s.io/HEAD/docs/proposals/custom-metrics.md#api
+type CustomMetricsEndpoint struct {
+	API   string   `json:"api",omitempty`
+	Path  string   `json:"path",omitempty`
+	Port  int      `json:"port"`
+	Names []string `json:"names"`
+}
+
+func AddCustomMetricsLabelIfRequired(pod *api.Pod, labels map[string]string) {
+	if _, ok := pod.Annotations[CustomMetricsAnnotationKey]; ok {
+		labels["io.cadvisor.metric.prometheus"] = CustomMetricsDefinitionContainerPath
+	}
+}
+
+func MakeCustomMetricsMountIfRequired(
```

Comment:
Currently, I have two public functions (AddCustomMetricsLabelIfRequired and MakeCustomMetricsMountIfRequired) which are not instance methods. Introducing an interface will require making them instance methods that need to be run on a dummy custom metrics package object (object w/o any instance variables). So, in every call, we will need to create an empty custom metric package object just to call a method on it. I don't think this will be cleaner, but if you insist I will do it.